src/js/views/patients/worklist/filters/date-filter_component.js

```diff
@@ -68,7 +99,7 @@ export default Component.extend({
     const datePicker = new DateFilterPickerComponent({
       ui: this.getView().$el,
       uiView: this.getView(),
-      state: state.pick('selectedDate', 'selectedMonth'),
+      state: state.pick('selectedDate', 'selectedMonth', 'dateType'),
```

Comment:
So I think `onClickDate()` will need to change to
`this.showDatePicker(this.getState().pick('selectedDate', 'selectedMonth', 'dateType'));`

and inside showDatePicker if the `dateType` changes we'll need to `this.showDatePicker({ dateType });` with the new dateType but without the current state _or_ if we want to be really fancy we'd only use the current state if the current `dateType` was re-chosen.  This is necessary so that you can change the type and then choose the same currently selected date.
Then all of the `setRelatveDate`, `setMonth`, `setDate`s need to take a second argument which is the `dateType` so that the `dateType` updates on the component state at the same time.  And we should test that `data-date-filter-region` can change to another labeling via the component.